### PR TITLE
feat: dotnet segment for .NET SDK display

### PR DESCRIPTION
**Pull request by tillig**
### Prerequisites

- [x] I have read and understand the `CONTRIBUTING` guide
- [x] The commit message follows the [conventional commits][cc] guidelines
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

### Description

New segment for .NET SDK version (or unsupported version) display.

Includes update for handling command execution errors so segments
can act differently based on exit codes. Using a custom error
type to make it testable rather than passing the OS error directly
to the segment.

[cc]: https://www.conventionalcommits.org/en/v1.0.0/#summary

### CODE REVIEW

File: docs/docs/segment-dotnet.md [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+```json
+{
+  "type": "dotnet",
+  "style": "powerline",
+  "powerline_symbol": "",
+  "foreground": "#000000",
+  "background": "#00ffff",
+  "properties": {
+    "prefix": "  "
```
Comment (JanDeDobbeleer) [inline docs/docs/segment-dotnet.md]: Can you add the textual utf32 representation of the icon? Like `\uF321` for example.

File: docs/docs/segment-dotnet.md [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+  }
+}
+```
+
+## Properties
+
+- display_version: `boolean` - display the active version or not; useful if all you need is an icon indicating `dotnet`
+  is present - defaults to `true`
+- unsupported_version_icon: `string` - text/icon that is displayed when the active .NET SDK version (e.g., one specified
+  by `global.json`) is not installed/supported - defaults to `⌧`
```
Comment (JanDeDobbeleer) [inline docs/docs/segment-dotnet.md]: Same comment as above, use the UTF32 character text representation

File: segment_dotnet_test.go [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+func bootStrapDotnetTest(args *dotnetArgs) *dotnet {
+	env := new(MockedEnvironment)
+	env.On("hasCommand", "dotnet").Return(args.enabled)
+	if args.unsupported {
+		err := &commandError{exitCode: 145}
+		env.On("runCommand", "dotnet", []string{"--version"}).Return("", err)
+	} else {
+		env.On("runCommand", "dotnet", []string{"--version"}).Return(args.version, nil)
+	}
+
```
Comment (JanDeDobbeleer) [inline segment_dotnet_test.go]: not a fan of the spaces here myself :-)

**Review by JanDeDobbeleer — CHANGES_REQUESTED** [review]

File: docs/docs/segment-dotnet.md (hunk 1 above)
Comment (tillig) [reply docs/docs/segment-dotnet.md]: I can throw another PR in to fix up the other docs using extended characters if that's interesting.

**Comment by JanDeDobbeleer** [comment]
Can you rebase? I can still merge this today (beware, lots of changes)

**Comment by tillig** [comment]
Already rebased! I think I caught it like a minute before your comment. ;)

**Pull request merged**